### PR TITLE
Issue 13 allow administrator with crud actions

### PR DESCRIPTION
**Pull request by khoa250200**
Please review my code before merge into develop

### CODE REVIEW

**Review by anhpn0712 — APPROVED** [review]

**Review by hongthamnguyen2k — APPROVED** [review]

File: app/assets/stylesheets/access_denied.scss [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#access-denied {
+font-family: 'Ropa Sans', sans-serif;
```
Comment (KhoaNDAHopee) [inline app/assets/stylesheets/access_denied.scss]: Please check indent

File: app/controllers/administrator/admins_controller.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module Administrator
+  class AdminsController < Administrator::BaseController
+    before_action :authorize_admin
+    before_action :set_admin, only: %i[show edit update destroy]
+
+    def index
+      @admins = Admin.all
```
Comment (KhoaNDAHopee) [inline app/controllers/administrator/admins_controller.rb:14]: Please use pagination

File: app/models/admin.rb [hunk 3]
```diff
@@ -3,12 +3,11 @@
...
   before_save { self.email = email.downcase }
   before_save { self.phone = phone.to_s }
-  # Include default devise modules. Others available are:
-  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
 
-  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  VALID_EMAIL_REGEX = /\A[^@\s]+@[^@\s]+\z/
```
Comment (KhoaNDAHopee) [inline app/models/admin.rb]: Try with ``devise::email_regexp``

File: spec/factories/admins.rb [hunk 4]
```diff
@@ -1,11 +1,13 @@
+# frozen_string_literal: true
+
 FactoryBot.define do
   factory :admin do
     first_name { Faker::Name.first_name }
     last_name { Faker::Name.last_name }
-    phone { Faker::PhoneNumber.cell_phone }
+    phone { '1111222233' }
```
Comment (KhoaNDAHopee) [inline spec/factories/admins.rb]: Why hard set for phone?

File: spec/factories/admins.rb [hunk 5]
```diff
@@ -1,11 +1,13 @@
...
     email { Faker::Internet.email }
-    status { [:active, :inactive].sample }
-    role { [:admin, :customer_service].sample }
+    status { 'active' }
```
Comment (KhoaNDAHopee) [inline spec/factories/admins.rb]: Why hard set for status?

File: spec/factories/admins.rb [hunk 6]
```diff
@@ -1,11 +1,13 @@
...
+    role { 'admin' }
```
Comment (KhoaNDAHopee) [inline spec/factories/admins.rb]: Why hard set for role?

File: db/seeds.rb [hunk 7]
```diff
@@ -9,6 +9,6 @@
 #
 #   movies = Movie.create([{ name: "Star Wars" }, { name: "Lord of the Rings" }])
 #   Character.create(name: "Luke", movie: movies.first)
-5.times do 
+1.times do 
   create(:admin)
```
Comment (KhoaNDAHopee) [inline db/seeds.rb]: If you want create admin with role: admin, status: active
https://devhints.io/factory_bot -> use with options
Screen Shot 2023-05-04 at 11 00 38
```
create(:admin, role: 'admin', status: 'active')
```

File: spec/requests/admins_controller.rb [hunk 8]
```diff
@@ -0,0 +1,199 @@
+# frozen_string_literal: true
```
Comment (KhoaNDAHopee) [inline spec/requests/admins_controller.rb]: Refactor this file try with https://devhints.io/factory_bot

File: spec/requests/admins_controller.rb [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+include AuthHelper
+
+RSpec.describe Administrator::AdminsController, type: :controller do
+  # admin with role admin
+  let(:admin) { create(:admin) }
+  let(:admin_attributes) { attributes_for(:admin) }
+
+  # admin with role customer_service
+  let(:admin_customer_service) do
+    Admin.create(
```
Comment (KhoaNDAHopee) [inline spec/requests/admins_controller.rb]: Use factory_bot

File: spec/requests/admins_controller.rb (hunk 8 above)
Comment (KhoaNDAHopee) [reply spec/requests/admins_controller.rb]: Add _spec to file name, rspec can't scan this file and run test.
Screen Shot 2023-05-04 at 11 07 01

File: spec/factories/admins.rb [hunk 10]
```diff
@@ -1,11 +1,13 @@
+# frozen_string_literal: true
```
Comment (KhoaNDAHopee) [inline spec/factories/admins.rb:1]: Update password field

**Review by KhoaNDAHopee — CHANGES_REQUESTED** [review]
Please check my comments:
Screen Shot 2023-05-04 at 11 13 12
- Please remove unnecessary items on left menu
Screen Shot 2023-05-04 at 11 14 40
- Please implement UI for CURD admin

**Pull request merged**